model/common_structures.smithy

```diff
@@ -59,3 +59,30 @@ structure Hits{
     _source: Document,
     fields: Document
 }
+
+
+structure RoleMapping{
+    hosts: Hosts
+    users: Users
+    reserved: Boolean
+    hidden: Boolean
+    backend_roles: BackendRoles
+    and_backend_roles: AndBackendRoles
+    description: String
+}
+
+structure RoleMappingResponse{
+    status: String
+    message: String
+}
+
+structure PatchOperation {
+    op: String,
+    path: String,
+    value: AttributeMap
+}
+
+map AttributeMap {
+    key: String,
+    value: String
+}
```

Comment:
This file will grown extremely large if we put every structure here. It's only meant for structure to be reused like the results of different types of search. Can we move these into the structures.smithy file of the respective operation?